File: app/src/main/java/com/util/base/statemachine/PersonMachine.java
package com.util.base.statemachine;

/**
 * 基于人的行为状态机
 */
public interface PersonMachine {
    PersonMachine INSTANCE = new PersonMachineImpl();

    void start();

    void pause();

    void resume();

    void stop();

    void sleep();

    void wake();

    void eat();

    void drive();

    void dream();
}

File: app/src/main/java/com/util/base/statemachine/PersonMachine.kt
package com.util.base.statemachine

/**
 * 基于人的行为状态机
 */
interface PersonMachine {
    fun start()
    fun pause()
    fun resume()
    fun stop()
    fun sleep()
    fun wake()
    fun eat()
    fun drive()
    fun dream()

    companion object {
        @JvmField
        val INSTANCE: PersonMachine = PersonMachineImpl()
    }
}